### PR TITLE
Use ballocated for tests

### LINKED ISSUE #89: Allocation tests post-1.3
Many of the allocation tests fail on master (and on 1.4), essentially because this is nonzero:
```
julia> @allocated (() -> (1+1))()
62992

julia> VERSION
v"1.5.0-DEV.88"
```
On slack, kristofferc points me to https://github.com/JuliaLang/julia/pull/33717, after which the `@allocated` macro no longer excludes compilation cost. So we will need to adjust how we test.

### PR DESCRIPTION
**Pull request by oxinabox**
Closes #89

### CODE REVIEW

File: test/wrapper_array.jl [hunk 1]
```diff
@@ -150,9 +150,10 @@ end
 
 const cnda = NamedDimsArray([10 20; 30 40], (:x, :y))
 @testset "allocations: wrapper" begin
-    @test 0 == @allocated parent(cnda)
-    @test 0 == @allocated dimnames(cnda)
+    @test 0 == @ballocated parent(cnda)
+    @test 0 == @ballocated dimnames(cnda)
 
+    # These tests use `@allocated` as for some reason `@ballocated` reports 1 alloc
     @test 0 == @allocated refine_names(cnda, (:x, :y))
```
Comment (oxinabox) [inline test/wrapper_array.jl]: This is a bit odd.
I think it relates to https://github.com/JuliaLang/julia/issues/34871

File: src/wrapper_array.jl [hunk 2]
```diff
@@ -35,7 +35,7 @@ struct NamedDimsArray{L, T, N, A<:AbstractArray{T, N}} <: AbstractArray{T, N}
     data::A
 end
 
-function NamedDimsArray{L}(orig::AbstractArray{T, N}) where {L, T, N}
+@inline function NamedDimsArray{L}(orig::AbstractArray{T, N}) where {L, T, N}
```
Comment (omus) [inline src/wrapper_array.jl]: Did this sneak in or is it related to allocations?

Comment (oxinabox) [reply src/wrapper_array.jl]: It is related to allocations.
Without this, it allocates

File: test/wrapper_array.jl (hunk 1 above)
Comment (nickrobinson251) [reply test/wrapper_array.jl]: https://github.com/JuliaLang/julia/issues/14955

**Comment by oxinabox** [comment]
Mac error is a Travis fault

**Pull request merged**